### PR TITLE
11 implement a button to start the creation of a route

### PR DESCRIPTION
**Pull request by changchuntzu0618**
The GPS and tracking is working. When press the start button it will start to locate your position and draw the route.

test reach 87% for map package

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/stepquest/map/MapTest.kt
Comment (Raytimz) [inline app/src/androidTest/java/com/github/se/stepquest/map/MapTest.kt:1]: You sure you need these comments?

File: app/src/main/java/com/github/se/stepquest/map/LocationViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    if (fusedLocationClient == null) {
+      fusedLocationClient = LocationServices.getFusedLocationProviderClient(context)
+    }
+    locationCallback =
+        object : LocationCallback() {
+          override fun onLocationResult(p0: LocationResult) {
+            for (lo in p0.locations) {
+              // Update UI with location data
+              currentLocation.value = LocationDetails(lo.latitude, lo.longitude)
+              //              println("Location in view: ${currentLocation.value}")
```
Comment (Raytimz) [inline app/src/main/java/com/github/se/stepquest/map/LocationViewModel.kt:37]: Are those prints were used for debug? Probably the pull request should be without them

File: app/src/main/java/com/github/se/stepquest/map/Map.kt [hunk 2]
```diff
@@ -1,14 +1,149 @@
...
+  return LatLng(latitude, longitude)
+}
+
+fun initMap(googleMap: GoogleMap) {
+  //    println("Initiate map")
+  googleMap.setMapType(GoogleMap.MAP_TYPE_HYBRID)
+  googleMap.uiSettings.isZoomControlsEnabled = true
+}
+
+// @Preview
```
Comment (Raytimz) [inline app/src/main/java/com/github/se/stepquest/map/Map.kt:145]: You don't this anymore I guess..

File: app/src/main/java/com/github/se/stepquest/map/Map.kt [hunk 3]
```diff
@@ -1,14 +1,149 @@
...
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+import com.google.android.gms.maps.model.PolylineOptions
 import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.ktx.addMarker
 
 @Composable
-fun Map() {
+fun Map(locationViewModel: LocationViewModel) {
   //  println("Here in Map")
```
Comment (Raytimz) [inline app/src/main/java/com/github/se/stepquest/map/Map.kt:40]: Another debug print?

**Review by Raytimz — CHANGES_REQUESTED** [review]
I think the code needs a bit more cleaning up. Also the updates from the main branch should be included.

File: app/src/main/java/com/github/se/stepquest/map/Map.kt (hunk 2 above)
Comment (changchuntzu0618) [reply app/src/main/java/com/github/se/stepquest/map/Map.kt]: yes, i delete it also thx

File: app/src/main/java/com/github/se/stepquest/map/LocationViewModel.kt (hunk 1 above)
Comment (changchuntzu0618) [reply app/src/main/java/com/github/se/stepquest/map/LocationViewModel.kt]: yes i delete it also

**Review by NizarBenMohamed — CHANGES_REQUESTED** [review]
Code and tests look good, however you need to clean up CI.yml to remove the new branch. Also you modified the premissions, should be changed back  for step sensor to work.

**Review by NizarBenMohamed — APPROVED** [review]

**Review by Raytimz — APPROVED** [review]
Code looks clean and coherent to me. The feature works on my end system.

**Pull request merged**